pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -702,6 +708,9 @@ func buildMasterMachineSet(clusterDeployment *clustop.ClusterDeployment, cluster
 	}
 	machineSet.Labels[clustop.ClusterDeploymentLabel] = clusterDeployment.Name
 	machineSet.Labels[clustop.ClusterNameLabel] = cluster.Name
+	for k, v := range machineSetConfig.NodeLabels {
```

Comment:
Same as above

Reply:
Am I correct that because this one is actually the labels on a master machine set being created, we would leave the door open for external actors setting labels, but a few lines down we would not. (comment incoming there)

Reply:
Ignore previous comment, this one and below look correct, we need to add the assumed labels for cluster name and deployment, and everything the user requested in NodeLabels, sound right?